### PR TITLE
Refactor `fromProtocol`

### PR DESCRIPTION
**Pull request by Jimbo4350**
This PR: 
- Refactors `fromProtocol` to accept exactly what it needs.
- Removes extraneous command line arguments as a result of the above refactor.

Relevant: #311

### CODE REVIEW

**Comment by CodiePP** [comment]
> The node should not determine its IP address from a topology file, it should use a topology file to determine its peer's IP addresses.

@denisshevchenko is working on issue: https://github.com/input-output-hk/cardano-node/issues/329 which should eliminate the need for a topology file.

File: cardano-config/src/Cardano/Config/CommonCLI.hs [hunk 1]
```diff
@@ -95,6 +99,14 @@ parseGenesisPathLast =
         <> help "The filepath to the genesis file."
     )
 
+parseGenesisHash :: Parser Text
```
Comment (CodiePP) [inline cardano-config/src/Cardano/Config/CommonCLI.hs]: looking forward to have this option on the command line

**Comment by Jimbo4350** [comment]
> this PR touches 32 files! can it be broken in smaller chunkes?
> and, there is work underway to remove the topology. It might make sense to push this other PRs first and then rebase.

I can cut it in half. I think @deepfire is still trying to get the chairman to work on #318 so I'd prefer to get this merged as I'm not sure how long that will take. It's not a big difference in terms of rebase conflict & I've reduced the PR by ~50%.

**Comment by Jimbo4350** [comment]
> The node should not determine its IP address from a topology file, it should use a topology file to determine its peer's IP addresses.

@karknu I've reduced this PR, so in a separate PR I will address this. You wanted to specify the IP via the command line correct?

**Comment by karknu** [comment]
> > The node should not determine its IP address from a topology file, it should use a topology file to determine its peer's IP addresses.
> 
> @karknu I've reduced this PR, so in a separate PR I will address this. You wanted to specify the IP via the command line correct?

Yeah, an optional command line argument for the IP address will work fine.

**Pull request merged**